jdbc/src/java/org/apache/hive/jdbc/HiveStatement.java

```diff
@@ -196,6 +197,14 @@ private void closeStatementIfNeeded() throws SQLException {
       }
     } catch (SQLException e) {
       throw e;
+    } catch (TApplicationException tae) {
+      String errorMsg = tae.getMessage();
```

Comment:
The default message should still be `Failed to close statement`.  The message of the `TApplicationException` will be preserved because the entire Exception is passed into the `SQLException`.

With that said, you can you please append that to the `BAD_SEQUENCE_ID` error as well?

> Failed to close statement.  Mismatch thrift sequence id. A previous ...

Reply:
Done